### PR TITLE
Add new MSAL Authorizer

### PR DESCRIPTION
**Pull request by lthela2**
WI: https://msazure.visualstudio.com/AzureArcPlatform/_workitems/edit/27877112

Added new MSAL AAD Authoizer

Tested this branch in this PR: https://msazure.visualstudio.com/AzureArcPlatform/_git/arcplatform/pullrequest/9986162

Appliance DF: https://msazure.visualstudio.com/AzureArcPlatform/_build/results?buildId=92536536&view=results
Custom Locations INT: https://msazure.visualstudio.com/AzureArcPlatform/_build/results?buildId=92532890&view=results

Logs: https://dataexplorer.azure.com/clusters/arcplatformdf.eastus2/databases/arcplatformservices?query=H4sIAAAAAAAAA42MuwrCMBRAd7%2FiktlBLIhLh9u0Tq0IxQ8I6aWNpknJTdXBj9cuvkBxOsM5HFlugneRXCMnXKIl5tkVRme8A1lmSh%2B%2FSBwGa5TTxL8Wz%2Bqv1bs9dxQImMLJaMpGYxtIUxAqWa0VLRNaiEfUcwudYhB7Nq6FqsYSEHPAMXbCh0%2BP%2Bau%2FX4bgD6Tj1M1hi1VR71AWN3zPUjIfAQAA

image

### CODE REVIEW

File: pkg/gencosmosdb/cosmosdb/collection.go [hunk 1]
```diff
@@ -180,6 +180,28 @@ type PartitionKeyRange struct {
 	Status             PartitionKeyRangeStatus `json:"status,omitempty"`
 	Parents            []string                `json:"parents,omitempty"`
 	LSN                int                     `json:"lsn,omitempty"`
+	MissingFields
```
Comment (julienstroheker) [inline pkg/gencosmosdb/cosmosdb/collection.go:183]: how is this related to this PR ?

File: example/cosmosdb/zz_generated_cosmosdb_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Code generated by github.com/julienstroheker/go-cosmosdb, DO NOT EDIT.
+
+package cosmosdb
```
Comment (julienstroheker) [inline example/cosmosdb/zz_generated_cosmosdb_suite_test.go:3]: How is this now generated ?

File: pkg/gencosmosdb/cosmosdb/collection.go (hunk 1 above)
Comment (lthela2) [reply pkg/gencosmosdb/cosmosdb/collection.go:183]: This change is from the Upstream. 
I am seeing the below error in E2Es with missing struct fields:

`example_test.go:131: json decode error [pos 355]: no matching struct field found when decoding stream map with key ownedArchivalPKRangeIds`

File: example/cosmosdb/zz_generated_cosmosdb_suite_test.go (hunk 2 above)
Comment (lthela2) [reply example/cosmosdb/zz_generated_cosmosdb_suite_test.go:3]: I think `make unittest` generated this file.

File: pkg/gencosmosdb/cosmosdb/collection.go (hunk 1 above)
Comment (julienstroheker) [reply pkg/gencosmosdb/cosmosdb/collection.go:183]: oh I remember this

**Review by julienstroheker — APPROVED** [review]
